crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -741,7 +741,7 @@ pub fn comparator_operand_type_rules(
 
             let comptime = CompTime::No(None);
             if other.try_bind_to_polymorphic_int(var, &comptime, true, op.location.span).is_ok() || other == &Type::Error {
-                Ok(other.clone())
+                Ok(Bool(comptime))
```

Comment:
A quick bugfix discovered by this PR. I can move it out but it would be blocking this PR and is a one line change that should not be controversial: `<` should always return a `bool`.